testsuite/openshift/objects/auth_config/sections.py

```diff
@@ -57,99 +58,85 @@ def add_item(
             item["priority"] = priority
         self.section.append(item)
 
+    @modify
+    def clear_all(self):
+        """Removes all identities from AuthConfig"""
+        self.section.clear()
 
-class Identities(Section):
+
+class IdentitySection(Section):
     """Section which contains identity configuration"""
 
     @modify
-    def mtls(self, name: str, selector_key: str, selector_value: str, **common_features):
+    def add_mtls(self, name: str, selector: Selector, **common_features):
         """Adds mTLS identity
         Args:
             :param name: name of the identity
-            :param selector_key: selector key to match
-            :param selector_value: selector value to match
+            :param selector: selector to match
         """
-        self.add_item(name, {"mtls": {"selector": {"matchLabels": {selector_key: selector_value}}}}, **common_features)
+        self.add_item(name, {"mtls": {"selector": asdict(selector)}, **common_features})
 
     @modify
-    def kubernetes(self, name: str, *audiences, **common_features):
+    def add_kubernetes(self, name: str, audiences: list[str], **common_features):
         """Adds Kubernetes identity
         Args:
             :param name: name of the identity
             :param audiences: token audiences
         """
-        self.add_item(name, {"kubernetes": {"audiences": [*audiences]}}, **common_features)
+        self.add_item(name, {"kubernetes": {"audiences": audiences}}, **common_features)
 
     @modify
-    def oidc(self, name, endpoint, credentials="authorization_header", selector="Bearer", **common_features):
+    def add_oidc(self, name, endpoint, *, credentials: Credentials = None, **common_features):
         """Adds OIDC identity"""
-        self.add_item(
-            name,
-            {"oidc": {"endpoint": endpoint}, "credentials": {"in": credentials, "keySelector": selector}},
-            **common_features
-        )
+        if credentials is None:
+            credentials = Credentials("authorization_header", "Bearer")
+        self.add_item(name, {"oidc": {"endpoint": endpoint}, "credentials": asdict(credentials)}, **common_features)
 
     @modify
-    def api_key(
+    def add_api_key(
         self,
         name,
+        *,
```

Comment:
why do you need a star here?

Reply:
I do not need it, I think it would make sense to require some arguments to be passed only as a keywords